### PR TITLE
Pgstac cql2text

### LINKED ISSUE #345: collections/{collection_id}/items and collections/{collection_id}/items/{item_id} return incorrect results
## Summary
Seeing incorrect results for collections/{collection_id}/items and collections/{collection_id}/items/{item_id}. Maybe this is a pgstac error?

## To Reproduce
add lines to create empty-collection at end of `ingest_joplin_data` function (right below line [44](https://github.com/stac-utils/stac-fastapi/blob/2.3.0/scripts/ingest_joplin.py#L45)):
```python
    collection["id"] = "empty-collection"
    post_or_put(urljoin(app_host, "/collections"), collection)
```

run pgstac app
```
git checkout 2.3.0
docker-compose build
docker-compose -d app-pgstac
python scripts/ingest_joplin.py http://localhost:8082
```

## Expected

- http://0.0.0.0:8082/collections/empty-collection/items
should return context { limit: 10, returned: 0}

- http://0.0.0.0:8082/collections/empty-collection/items/fe916452-ba6f-4631-9154-c249924a122d
should return return 404

- http://0.0.0.0:8082/collections/joplin/items/f7f164c9-cfdf-436d-a3f0-69864c38ba2a
should return item for id f7f164c9-cfdf-436d-a3f0-69864c38ba2a

## Actual results
- http://0.0.0.0:8082/collections/empty-collection/items
returns first 10 items for joplin

- http://0.0.0.0:8082/collections/empty-collection/items/fe916452-ba6f-4631-9154-c249924a122d
returns joplin item fe916452-ba6f-4631-9154-c249924a122d

- http://0.0.0.0:8082/collections/joplin/items/f7f164c9-cfdf-436d-a3f0-69864c38ba2a
returns joplin item fe916452-ba6f-4631-9154-c249924a122d

### PR DESCRIPTION
**Pull request by bitner**
**Related Issue(s):** #
Fixes #345 caused by regression in PGStac

**Description:**
Bump PGStac version to 0.4.5
  - Fixes issues with using direct parameters (ids, collections, datetime, intersects) with cql2-json
Adds support for cql2-text
  - Use PyGeoFilter to convert cql2-text on a get request into cql2-json to add support for cql2-text

**PR Checklist:**

- [x] Code is formatted and linted (run `pre-commit run --all-files`)
- [x] Tests pass (run `make test`)
- [x] Documentation has been updated to reflect changes, if applicable, and docs build successfully (run `make docs`)
- [x] Changes are added to the [CHANGELOG](https://github.com/stac-utils/pystac/blob/master/CHANGES.md).

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -32,9 +32,9 @@ services:
   app-pgstac:
     container_name: stac-fastapi-pgstac
     image: stac-utils/stac-fastapi
-    build:
-      context: .
-      dockerfile: Dockerfile
+    # build:
+    #   context: .
+    #   dockerfile: Dockerfile
```
Comment (lossyrob) [inline docker-compose.yml:37]: Uncomment.

File: stac_fastapi/pgstac/setup.py [hunk 2]
```diff
@@ -16,6 +16,7 @@
     "asyncpg",
     "buildpg",
     "brotli_asgi",
+    "pygeofilter @ git+https://github.com/geopython/pygeofilter@main#egg=pygeofilter",
```
Comment (lossyrob) [inline stac_fastapi/pgstac/setup.py:19]: Do we have a timeline for release?
If not, we should at least pin this to a commit.

File: stac_fastapi/pgstac/stac_fastapi/pgstac/core.py [hunk 3]
```diff
@@ -265,6 +269,15 @@ async def get_search(
         Returns:
             ItemCollection containing items which match the search criteria.
         """
+        request = kwargs["request"]
+        query_params = str(request.query_params)
+
+        # Kludgy fix because using factory does not allow alias for filter-lang
```
Comment (lossyrob) [inline stac_fastapi/pgstac/stac_fastapi/pgstac/core.py:275]: Can't follow this comment. Can you clarify what factor this would use ideally?

**Review by lossyrob — CHANGES_REQUESTED** [review]

File: stac_fastapi/pgstac/setup.py (hunk 2 above)
Comment (bitner) [reply stac_fastapi/pgstac/setup.py]: I don't know timeline for release, but these changes are in the tagged version v0.1.1

**Review by lossyrob — APPROVED** [review]

**Pull request merged**